src/main/java/org/mqttbee/mqtt/MqttClientData.java

```diff
@@ -181,6 +224,19 @@ public void setClientConnectionData(@Nullable final MqttClientConnectionData cli
         return Optional.of(serverConnectionData);
     }
 
+    @NotNull
+    @Override
+    public Optional<MqttClientSslData> getSslData() {
+        MqttClientSslData sslData;
```

Comment:
All this can be removed if the MqttClientSslConfig is passed as a argument to the constructor and then only returned here

Reply:
If this is done within the MqttClientSslConfigBuilder.build then the user also gets feedback if there is something wrong with his keystore...

Reply:
this one is still open. If the other ones are now ok, could maybe do this in a later commit? Perhaps we could then also discuss the impl.

Reply:
If we return here the SslConfig (which replaces/renames the SslData) and have a SslConfigBuilder the creation of the SslConfig takes place with the build() method before it is passed to the ClientBuilder